docs/updater/updater.md

```diff
@@ -25,3 +25,7 @@ For the above listed reasons, we cannot simply use TUF's updater like described
 - New metadata files (not counting delegated roles) are acquired by calling TUF's `refresh`. Metadata files belonging to the delegated roles are updated by `get_current_targets`. Information about a target file is obtained using `get_one_valid_targetinfo`
 
 - Since TUF does not provide a way to gradually update metadata and target files (except for `root.json`), that has to be handled by TAF. So, TAF's updater calls TUF's updater multiple times. Our handler, `GitUpdateHandler` finds all commits that were pushed after the last time the client pulled the changes. The basic idea is to traverse trough these commits and update metadata files and targets for each of them, thus validating all revisions.
+
+- Validation and syncing before running the updater is crucial for the updater to work. Using `git pull`, we can ensure that both the authentication repository and target repositories are aligned with their respective last validated states. The process involves comparing the current state of the repositories on the client's machine with the latest state on the remote server and making necessary adjustments to bring them into sync. Examples of when validation and syncing is necessary when the client's authentication repository is not in sync with the remote repository, when target repositories are out of sync, when there are invalid or unauthorized commits, when repositories are in mixed states, and after manual updates. 
+
+- Validating the last validated commit in the updater involves ensuring that the commit being used as a reference for updates is still valid and consistent with the current state of the repository. The process starts by identifying the main branch of the repository and obtaining the latest commit on that branch. If no specific commit is provided for validation, the process defaults to using the first commit in the branch's history. Next, the updater checks whether the provided last validated commit is still part of the branch's commit history. If the commit is missing, the updater fetches the latest commits from the remote repository to verify if the commit exists there. If the commit is found but the repository's current state has moved beyond this commit without aligning properly, the updater ensures that the last validated commit is updated to match the latest commit from the remote repository, guaranteeing that the validation process is based on the most recent and accurate state of the repository.
```

Comment:
Same here's we should never set the last validated commit to a commit that has not yet been fetched. If th Last validated commit is already set to a commit that is not contained by the local repo, but is by remote one, we don't have to raise an error.